clients/bmp_client.go

```diff
@@ -79,3 +79,181 @@ func (bc *bmpClient) SlPackages() (SlPackagesResponse, error) {
+		errorMessage := fmt.Sprintf("bmp: could not calls /sl/packages/'%s'/options on BMP server, error message '%s'", packageId, err.Error())
+		return SlPackageOptionsResponse{}, errors.New(errorMessage)
+	}
+
+	if slcommon.IsHttpErrorCode(errorCode) {
+		errorMessage := fmt.Sprintf("bmp: could not call /sl/packages/'%s'/options on BMP server, HTTP error code: '%d'", packageId, errorCode)
+		return SlPackageOptionsResponse{}, errors.New(errorMessage)
+	}
+
+	response := SlPackageOptionsResponse{}
+
+	err = json.Unmarshal(responseBytes, &response)
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: failed to decode JSON response, err message '%s'", err.Error())
+		return SlPackageOptionsResponse{}, errors.New(errorMessage)
+	}
+
+	return response, nil
+}
+
+func (bc *bmpClient) Tasks(latest int) (TasksResponse, error) {
+	path := fmt.Sprintf("%s/%s%d", bc.url, "/tasks?latest=", latest)
+	responseBytes, errorCode, err := bc.httpClient.DoRawHttpRequest(path, "GET", &bytes.Buffer{})
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: could not calls /tasks?latest='%d'on BMP server, error message '%s'", latest, err.Error())
+		return TasksResponse{}, errors.New(errorMessage)
+	}
+
+	if slcommon.IsHttpErrorCode(errorCode) {
+		errorMessage := fmt.Sprintf("bmp: could not call /tasks?latest='%d' on BMP server, HTTP error code: '%d'", latest, errorCode)
+		return TasksResponse{}, errors.New(errorMessage)
+	}
+
+	response := TasksResponse{}
+
+	err = json.Unmarshal(responseBytes, &response)
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: failed to decode JSON response, err message '%s'", err.Error())
+		return TasksResponse{}, errors.New(errorMessage)
+	}
+
+	return response, nil
+}
+
+func (bc *bmpClient) TaskOutput(taskId int, level string) (TaskOutputResponse, error) {
+	path := fmt.Sprintf("%s/task/%d/txt/%s", bc.url, taskId, level)
+	responseBytes, errorCode, err := bc.httpClient.DoRawHttpRequest(path, "GET", &bytes.Buffer{})
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: could not calls /task/'%d'/txt/'%s'on BMP server, error message '%s'", taskId, level, err.Error())
+		return TaskOutputResponse{}, errors.New(errorMessage)
+	}
+
+	if slcommon.IsHttpErrorCode(errorCode) {
+		errorMessage := fmt.Sprintf("bmp: could not call /task/'%d'/txt/'%s' on BMP server, HTTP error code: '%d'", taskId, level, errorCode)
+		return TaskOutputResponse{}, errors.New(errorMessage)
+	}
+
+	response := TaskOutputResponse{}
+
+	err = json.Unmarshal(responseBytes, &response)
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: failed to decode JSON response, err message '%s'", err.Error())
+		return TaskOutputResponse{}, errors.New(errorMessage)
+	}
+
+	return response, nil
+}
+
+func (bc *bmpClient) UpdateStatus(serverId string, status string) (UpdateStatusResponse, error) {
+	path := fmt.Sprintf("%s/baremetal/%s/%s", bc.url, serverId, status)
+	responseBytes, errorCode, err := bc.httpClient.DoRawHttpRequest(path, "PUT", &bytes.Buffer{})
```

Comment:
Hi dr.max,
I checked the ruby bmp client, the PUT request for update status doesn't take request body ,either.  The desired status and target server id are all specified in the url.

URI("#{config.target}/baremetal/#{server_id}/#{state}")